examples/network/commst.rs

```diff
@@ -59,16 +48,19 @@ impl<'a, M: Serialize + DeserializeOwned + Send + 'a> CommsTask<'a, M> {
 
     /// The main socket IO loop and an asynchronous thread responding to manager
     /// thread requests.
-    pub fn run(mut self) -> Result<(), Error> {
+    pub fn run(mut self) -> Result<(), Box<dyn std::any::Any + Send + 'static>> {
         // Borrow parts of `self` before entering the thread binding scope.
         let tx = self.tx;
         let rx = self.rx;
-        let mut stream1 = self.stream.try_clone()?;
+        let mut stream1 = match self.stream.try_clone() {
+            Ok(stream) => stream,
+            Err(e) => return Err(Box::new(e)),
+        };
```

Comment:
Would this work, too? (Or even without the `map_err` because `Box` implements `From`?)
```rust
let mut stream1 = match self.stream.try_clone().map_err(Box::new)?;
```

Reply:
I tried those before and it didn't work. Pattern-matching is the best I could come up with.